### PR TITLE
bug fix #4263

### LINKED ISSUE #4623: AutoComplete: scroll is not working when using down key in grouped mode
### Describe the bug

The scroll does not follow the highlighted item when using the down key to navigate through the suggested list.

P.S. The attached CodeSandbox link is from your documentation page.

### Reproducer

https://codesandbox.io/s/hn7jpp

### PrimeReact version

9.6.0

### React version

18.x

### Language

TypeScript

### Build / Runtime

Vite

### Browser(s)

Chrome 114.0.5735.199(Official Build) (64-bit)

### Steps to reproduce the behavior

1. Press "a" in the grouped autocomplete.
2. By using the down key to navigate to the bottom of the result panel, you will notice that the scroll is not following the focused item.

### Expected behavior

_No response_

### PR DESCRIPTION
**Pull request by userkks**
Fix #4623 : scroll container was passed wrong

### CODE REVIEW

File: components/lib/autocomplete/AutoComplete.js [hunk 1]
```diff
@@ -446,7 +446,7 @@ export const AutoComplete = React.memo(
         };
 
         const getScrollableElement = () => {
-            return virtualScrollerRef.current ? overlayRef.current.firstChild : overlayRef.current;
+            return virtualScrollerRef.current ? overlayRef.current.firstChild : overlayRef.current.firstChild;
```
Comment (melloware) [inline components/lib/autocomplete/AutoComplete.js:449]: This makes both sides of the ternary the same and will break other scenarios.  How about...
```suggestion
            return virtualScrollerRef.current || props.optionGroupLabel ? overlayRef.current.firstChild : overlayRef.current;
```

**Review by melloware — CHANGES_REQUESTED** [review]

File: components/lib/autocomplete/AutoComplete.js (hunk 1 above)
Comment (userkks) [reply components/lib/autocomplete/AutoComplete.js]: Hi @melloware, you are right the both side is same for the ternary operator. But the issue is there with all kind of autocompletes not only grouped. Please check my latest changes.

Comment (melloware) [reply components/lib/autocomplete/AutoComplete.js]: let me test in the showcase.

Comment (melloware) [reply components/lib/autocomplete/AutoComplete.js]: Yep your fix seems to be working for all the showcase examples

**Review by melloware — APPROVED** [review]

**Pull request merged**